src/Compilers/CSharp/Test/Semantic/IOperation/IOperationTests_IInterpolatedStringExpression.cs

```diff
@@ -444,5 +444,635 @@ public void M(int x)
+          Value: 
+            IParameterReferenceOperation: b (OperationKind.ParameterReference, Type: System.String) (Syntax: 'b')
+
+    Jump if False (Regular) to Block[B3]
+        IParameterReferenceOperation: a (OperationKind.ParameterReference, Type: System.Boolean) (Syntax: 'a')
+
+    Next (Regular) Block[B2]
+Block[B2] - Block
+    Predecessors: [B1]
+    Statements (1)
+        IFlowCaptureOperation: 1 (OperationKind.FlowCapture, Type: null, IsImplicit) (Syntax: 'b')
+          Value: 
+            IParameterReferenceOperation: b (OperationKind.ParameterReference, Type: System.String) (Syntax: 'b')
+
+    Next (Regular) Block[B4]
+Block[B3] - Block
+    Predecessors: [B1]
+    Statements (1)
+        IFlowCaptureOperation: 1 (OperationKind.FlowCapture, Type: null, IsImplicit) (Syntax: 'c')
+          Value: 
+            IParameterReferenceOperation: c (OperationKind.ParameterReference, Type: System.String) (Syntax: 'c')
+
+    Next (Regular) Block[B4]
+Block[B4] - Block
+    Predecessors: [B2] [B3]
+    Statements (1)
+        IExpressionStatementOperation (OperationKind.ExpressionStatement, Type: null, IsInvalid) (Syntax: 'b = $""{(a ? ...  c),d:D3}"";')
+          Expression: 
+            ISimpleAssignmentOperation (OperationKind.SimpleAssignment, Type: System.String, IsInvalid) (Syntax: 'b = $""{(a ? ... : c),d:D3}""')
+              Left: 
+                IFlowCaptureReferenceOperation: 0 (OperationKind.FlowCaptureReference, Type: System.String, IsImplicit) (Syntax: 'b')
+              Right: 
+                IInterpolatedStringOperation (OperationKind.InterpolatedString, Type: System.String, IsInvalid) (Syntax: '$""{(a ? b : c),d:D3}""')
+                  Parts(1):
+                      IInterpolationOperation (OperationKind.Interpolation, Type: null, IsInvalid) (Syntax: '{(a ? b : c),d:D3}')
+                        Expression: 
+                          IFlowCaptureReferenceOperation: 1 (OperationKind.FlowCaptureReference, Type: System.String, IsImplicit) (Syntax: 'a ? b : c')
+                        Alignment: 
+                          IParameterReferenceOperation: d (OperationKind.ParameterReference, Type: System.Int32, IsInvalid) (Syntax: 'd')
+                        FormatString: 
+                          ILiteralOperation (OperationKind.Literal, Type: System.String, Constant: ""D3"", IsInvalid) (Syntax: ':D3')
+
+    Next (Regular) Block[B5]
+Block[B5] - Exit
+    Predecessors: [B4]
+    Statements (0)
+";
+            var expectedDiagnostics = new DiagnosticDescription[] {
+                // file.cs(8,28): error CS0150: A constant value is expected
+                //         b = $"{(a ? b : c),d:D3}";
+                Diagnostic(ErrorCode.ERR_ConstantExpected, "d").WithLocation(8, 28)
+            };
+
+            VerifyFlowGraphAndDiagnosticsForTest<BlockSyntax>(source, expectedFlowGraph, expectedDiagnostics);
+        }
```

Comment:
>} [](start = 8, length = 1)

I specifically asked to "cover control flow **in each part where it can go**, even in error cases." I believe all test covering control flow were having it in an expression part. Is this the only place it can go, even in error cases?

Reply:
I don't understand the question/suggestion. Can you please clarify with a source example?

---
In reply to: [185934610](https://github.com/dotnet/roslyn/pull/26545#discussion_r185934610) [](ancestors = 185934610)

Reply:
> I don't understand the question/suggestion. Can you please clarify with a source example?

Can we have an expression with a control flow in alignment and format parts of an interpolation?

---
In reply to: [185935241](https://github.com/dotnet/roslyn/pull/26545#discussion_r185935241) [](ancestors = 185935241,185934610)

Reply:
Sure, I can add tests for that.

---
In reply to: [185935843](https://github.com/dotnet/roslyn/pull/26545#discussion_r185935843) [](ancestors = 185935843,185935241,185934610)

Reply:
Conditional flow is only possible in the alignment part, format part is always parsed as a string literal, so I will add tests for conditional code in alignment.

---
In reply to: [185936082](https://github.com/dotnet/roslyn/pull/26545#discussion_r185936082) [](ancestors = 185936082,185935843,185935241,185934610)

Reply:
> Conditional flow is only possible in the alignment part, format part is always parsed as a string literal, so I will add tests for conditional code in alignment.

Sounds good

---
In reply to: [185949535](https://github.com/dotnet/roslyn/pull/26545#discussion_r185949535) [](ancestors = 185949535,185936082,185935843,185935241,185934610)